web/src/components/RTLCacheProvider.jsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+import rtlPlugin from "stylis-plugin-rtl";
+import { CacheProvider } from "@emotion/react";
+import createCache from "@emotion/cache";
+import { prefixer } from "stylis";
+import { useTranslation } from "react-i18next";
+
+// https://mui.com/material-ui/guides/right-to-left
+
+const cacheRtl = createCache({
+  key: "muirtl",
+  stylisPlugins: [prefixer, rtlPlugin],
+});
+
+const RTLCacheProvider = ({ children }) => {
+  const { i18n } = useTranslation();
+
+  return i18n.dir() === "rtl" ? <CacheProvider value={cacheRtl}>{children}</CacheProvider> : children;
+};
+
+export default RTLCacheProvider;
```

Comment:
I know the guide says "use a cache provider", but i couldn't figure out why. Do you happen to know the answer?

Reply:
It's for caching the results of the CSS plugins (vendor autoprefixing, generated class names). Theres's a default cache  too; this one adds on the RTL plugin to the default plugins and then uses a separate cache.

Reply:
No idea what that means, hehe, but I trust you. :-D